### PR TITLE
Fixes #14101 - Exception triggered when editing/adding a role

### PR DESCRIPTION
**Pull request by dhlavac**

### CODE REVIEW

**Comment by ares** [comment]
[test]

**Comment by ares** [comment]
[test]

**Comment by ares** [comment]
Could you please add some reproducing steps? I failed to reproduce. I created a user which had all unlimited location permissions and permission to edit roles. I tried to let users without any locations as well as associate all. Roles editing and creation work just fine.

File: app/services/association_authorizer.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@ def self.authorized_associations(associations, klass_name = nil, should_raise_ex
     if associations.included_modules.include?(Authorizable)
       associations_klass = associations
       if associations.respond_to?(:klass)
-        associations_klass = associations.klass
+        if associations.klass == Taxonomy
```
Comment (ares) [inline app/services/association_authorizer.rb]: I was finally able to reproduce. After more digging I don't think this is the right place to fix it. When you check the backtrace you can see that this is called from https://github.com/theforeman/foreman/blob/105917c2f0cb2ef1200f2c5f12a304321b4f326d/app/helpers/form_helper.rb#L74 and that this method has optional argument called `klass_name`. The only thing that we need to do is passing right class when multiple_selects is used to generate taxonomies field. Something like this should work

``` ruby
def multiple_selects(f, attr, associations, selected_ids, options = {}, html_options = {})
    options.merge!(:size => "col-md-10")
    klass = case attr
              when :organizations
                Organization
              when :locations
                Location
              else
                nil
            end
    authorized = AssociationAuthorizer.authorized_associations(associations, klass).all
    ....
```

**Comment by dhlavac** [comment]
@ares something else to do ?

**Comment by ares** [comment]
I guess the test is not unrelated and can be removed, looks good otherwise.

**Comment by dhlavac** [comment]
Test removed

**Comment by dLobatog** [comment]
Merged as 8fde28895ec88297e88413e86ea191828bfabc7c, thanks @dhlavac!

**Pull request closed without merging**